### PR TITLE
Setup semantic release

### PR DESCRIPTION
**Pull request by andrii-balitskyi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,14 @@
 {
-  "name": "sdk1",
-  "version": "1.0.0",
-  "main": "index.js",
+  "name": "@seamapi/php",
+  "version": "2.0.0",
```
Comment (razor-x) [inline package.json:3]:
```suggestion
  "version": "2.0.0",
  "private": true,
```

File: .github/workflows/version.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+          git_committer_name: ${{ secrets.GIT_USER_NAME }}
+          git_committer_email: ${{ secrets.GIT_USER_EMAIL }}
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+      - name: Setup
+        uses: ./.github/actions/setup
+      - name: Cut ${{ github.event.inputs.version }} version
+        run: npm version --sign-git-tag=true ${{ github.event.inputs.version }}
```
Comment (razor-x) [inline .github/workflows/version.yml:35]: Ideally we could use some php tool here like composer and not rely on npm, but this is a good step in that direction. We can do this for now and replace it later. See for example how bundler and poetry handle this in our ruby and python repos

File: .github/workflows/generate.yml [hunk 3]
```diff
@@ -29,10 +29,6 @@ jobs:
           passphrase: ${{ secrets.GPG_PASSPHRASE }}
       - name: Setup
         uses: ./.github/actions/setup
-        with:
-          install_dependencies: "false"
-      - name: Normalize package-lock.json
-        run: npm install
```
Comment (razor-x) [inline .github/workflows/generate.yml:35]: Why remove this?

File: .github/actions/setup/action.yml
Comment (razor-x) [inline .github/actions/setup/action.yml:1]: Are these changes just ported from the newer version in the other repo?

Comment (andrii-balitskyi) [reply .github/actions/setup/action.yml:1]: If you're talking about https://github.com/seamapi/generated-python then no. That forked repo was our playground for getting the generation workflow and code testing sorted. Now that we've merged all the good stuff here in, I think we can archive that forked one.

Did I remove something important here? I'm still getting the hang of GitHub actions and workflows, so if you spot something's off or have any tips, I'm all ears!

File: .github/workflows/generate.yml (hunk 3 above)
Comment (andrii-balitskyi) [reply .github/workflows/generate.yml:35]: I thought we don't need it since `action.yml` gets the packages installed. Lmk if I got it wrong.

Comment (razor-x) [reply .github/workflows/generate.yml:35]: This fixes a know upstream issue with npm where different npm versions can create drift in the lockfile.

Let's use these workflow files unchanged as much as possible aside from what we need for php to work https://github.com/seamapi/javascript-http/tree/main/.github

**Review by razor-x — APPROVED** [review]

File: .github/workflows/generate.yml (hunk 3 above)
Comment (andrii-balitskyi) [reply .github/workflows/generate.yml]: https://github.com/seamapi/php/pull/95/commits/b45cf9203025dc6b677843362835b36095a1b480

**Pull request merged**